### DISCUSSION TITLE
How to get JAX to maximize GPU usage?

### DISCUSSION THREAD

**Discussion by cool-RR** [jax-ml/jax#22859] [Q&A]
Hey,

This is a newbie question about using JAX on GPUs because I'm new at using GPUs for anything. I noticed that when I run JAX experiments using GPUs (on a `g2-standard-8` on GCP), the GPU usage as shown by `nvtop` is around 60%, even when the calculations are taking a long time. Why is that not 100%? I've only run computations on CPUs before, where they usually go to 100%. Is there some kind of optimization that I would need to do to get JAX to use 100% of the GPU?

Thanks,
Ram.

**Comment by ddrous** [comment]
Hi Ram,

The CPU model is quite different from the massively parallel GPU. [On GPU](https://enterprise-support.nvidia.com/s/article/Useful-nvidia-smi-Queries-2), 60% represents the proportion of time your CUDA cores are busy executing a JAX kernel.

You say the calculations are taking a long time, do you know whether your program is fundamentally compute or memory bound ? Depending on that, you could improve on 60% by having some optimisations. The easiest things to try could be to `jit`-compile, making the problem bigger, improving data transfer between disk/RAM and GPU memory...

For now It'd be good to have a minimum working example of the program you are running, as well as the output of `jax.print_environment_info()`.

Cheers,
